src/frontend/lib/utils.ts

```diff
@@ -305,3 +305,29 @@ export function matchPreset(
 
   return bestMatch;
 }
+
+export function convertObservationsToFeatures(
```

Comment:
nit: ideally use [turf](https://turfjs.org/docs/api/points) for this

Reply:
I tried, but turf actually makes it slightly more complicated. We add `properties.id` to each feature (used for navigation). this function creates a feature collections, but then we need to iterate through it again to do add these ids

Reply:
yeah once again, should've done a better job of pointing to the appropriate docs. in this case, you can use the [`featureCollection`](https://turfjs.org/docs/api/featureCollection) and [`point`](https://turfjs.org/docs/api/point) helpers.

the resulting code would look similar to this:

```ts
import {featureCollection, point} from '@turf/helpers';

function mapObservationsToFeatures(observations: Observation[]) {
  // Ideally use a for-loop for performance reasons as opposed to looping twice, since observations array can get large
  const points = observations
    .filter(obs => typeof obs.lon === 'number' && typeof obs.lat === 'number')
    .map(obs =>
      point(
        // @ts-expect-error TS doesn't respect filtering properly (although fixed in TS 5.5)
        [obs.lon, obs.lat],
        {id: obs.docId},
      ),
    );

  return featureCollection(points);
}
```